src/freenet/clients/http/DiagnosticToadlet.java

```diff
@@ -409,15 +414,60 @@ public int compare(PeerNodeStatus firstNode, PeerNodeStatus secondNode) {
 		}
 		textBuilder.append("\n");
 
-		// drawThreadPriorityStatsBox
-		textBuilder.append("Threads:\n");
-		int[] activeThreadsByPriority = stats.getActiveThreadsByPriority();
-		int[] waitingThreadsByPriority = stats.getWaitingThreadsByPriority();
-		for(int i=0; i<activeThreadsByPriority.length; i++) {
-			textBuilder.append(l10n("running")).append(": ").append(String.valueOf(activeThreadsByPriority[i])).append(" (").append(String.valueOf(i+1)).append(")\n");
-			textBuilder.append(l10n("waiting")).append(": ").append(String.valueOf(waitingThreadsByPriority[i])).append(" (").append(String.valueOf(i+1)).append(")\n");
-		}
-		textBuilder.append("\n");
+			// drawThreadPriorityStatsBox
+			textBuilder.append("Threads:\n");
+			ThreadMXBean threadMxBean = ManagementFactory.getThreadMXBean();
+			Map<Long, Long> threadCPU = new HashMap<Long, Long>();
+			for (ThreadInfo info : threadMxBean.dumpAllThreads(false, false)) {
+			    threadMxBean.getThreadInfo(info.getThreadId());
+				threadCPU.put(info.getThreadId(), threadMxBean.getThreadCpuTime(info.getThreadId()));
+			}
+
+			RuntimeMXBean runtimeMxBean = ManagementFactory.getRuntimeMXBean();
+			long initialUptime = runtimeMxBean.getUptime();
+
+			try {Thread.sleep(1000);} catch (InterruptedException e) {}
```

Comment:
At least put this in its own method and format it correctly…

Reply:
Alternatively I was thinking launching a thread to keep this metrics up to date and not having to wait for this on every request. Definitely needs some more work.

Reply:
Yeah, doing that as its own component, independent of this page, that would be cool, a lot of stuff from this could go into that and not clutter up this code. :)

Reply:
I'll look into that. It'd avoid any cluttering and make it easier to remove / adapt if we need to. Any example components to look at for inspiration?

Reply:
I don’t think we have something like that, or, well, we probably have a lot of stuff that does something like that and is reachable from `Node` somehow but we never thought of it as components. Maybe something like `ThreadStats` in `freenet.node.stats`? That’s where I would put it unless I find something better while doing that. :)

Reply:
Nice one. I'll start from that and I'll see if there's a better spot. 👍

Reply:
Done. Created `NodeDiagnostics` as a module to add future metrics. For now it only carries `ThreadMonitor`.

- For now I choose to leave the sleep/queuedTimedJob intervals hardcoded but they could easily be moved into a configuration.
- There's some syncronization/thread safety going on with `nodeThreadInfos` that I'm not sure I'm doing right.
- I corrected the % CPU calculation, here's the reference I used: https://gist.github.com/lpsandaruwan/f2cef0aa91ae68cb041c7ecda04a0724#file-jvmcpuusage-java-L78